interaction.go

```diff
@@ -0,0 +1,104 @@
+package disgord
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type InteractionType = int
+
+const (
+	_ InteractionType = iota
+	InteractionPing
+	InteractionApplicationCommand
+	InteractionMessageComponent
+)
+
+type OptionType = int
+
+const (
+	_ OptionType = iota
+	SUB_COMMAND
+	SUB_COMMAND_GROUP
+	STRING
+	INTEGER
+	BOOLEAN
+	USER
+	CHANNEL
+	ROLE
+	MENTIONABLE
+)
+
+type InteractionCallbackType = int
+
+const (
+	_ InteractionCallbackType = iota
+	Pong
+	_
+	_
+	ChannelMessageWithSource
+	DeferredChannelMessageWithSource
+	DeferredUpdateMessage
+	UpdateMessage
+)
+
+//TODO ApplicationCommandInteractionDataResolved https://discord.com/developers/docs/interactions/slash-commands#interaction-applicationcommandinteractiondataresolved
+type ApplicationCommandInteractionDataResolved struct {
+}
+
+type ApplicationCommandInteractionDataOption struct {
+	Name    string                                     `json:"name"`
+	Type    OptionType                                 `json:"type"`
+	Options []*ApplicationCommandInteractionDataOption `json:"options"`
+}
+
+type ApplicationCommandInteractionData struct {
+	ID       Snowflake                                    `json:"id"`
+	Name     string                                       `json:"name"`
+	Resolved []*ApplicationCommandInteractionDataResolved `json:"resolved"`
+	Options  []*ApplicationCommandInteractionDataOption   `json:"options"`
+	CustomID string                                       `json:"custom_id"`
+	Type     MessageComponentType                         `json:"component_type"`
+}
+
+type MessageInteraction struct {
+	ID   Snowflake       `json:"id"`
+	Type InteractionType `json:"type"`
+	Name string          `json:"name"`
+	User *User           `json:"user"`
+}
+
+type InteractionApplicationCommandCallbackData struct {
+	Tts             bool             `json:"tts"`
+	Content         string           `json:"content"`
+	Embeds          []*Embed         `json:"embeds"`
+	Flags           int              `json:"flags"`
+	AllowedMentions *AllowedMentions `json:"allowed_mentions"`
+}
+
+type InteractionResponse struct {
+	Type InteractionCallbackType                    `json:"type"`
+	Data *InteractionApplicationCommandCallbackData `json:"data"`
+}
+
+func SendInteractionResponse(interaction *InteractionCreate, data *InteractionResponse) error {
```

Comment:
This needs to be a part of the client:
 1. rate limiting, if there is any
 2. re-use the http instance (proxy support + other things)

The way it's done in disgord now is a bit, over-engineered, and lately ive been trying to go away from it. However, see if you can create a `httd.Request` object and call `Client.req.Do(context.Context, *httd.Request)`. After the `Do` call, you don't need to close the body of the request and you can read directly from the returned body instead. Also no need to worry about error handling as this method does that for you.

Reply:
Done